### PR TITLE
BB 2.6.2, HA base 16.3.4

### PR DESCRIPTION
**Pull request by OttPeterR**

### CODE REVIEW

File: babybuddy/Dockerfile [hunk 1]
```diff
@@ -49,13 +49,16 @@ RUN \
...
-  pip && \
-  pip install -U --ignore-installed -r requirements.txt
+# Define PATH_MAX before including necessary headers
+RUN echo '#define PATH_MAX 4096' >> /usr/include/limits.h
 
+# Install dependencies and pandas
+RUN echo "**** installing ****" && \
+    cd /app/babybuddy && \
+    pip3 install -U --no-cache-dir pip && \
+    pip install --prefer-binary --no-cache-dir pandas && \
```
Comment (cdubz) [inline babybuddy/Dockerfile]:
```suggestion
```

Pandas is no longer a dependency. It looks like the latest versions do not support 32bit arch so this is why those arches are failing. Give this a try without this line.

**Review by cdubz — CHANGES_REQUESTED** [review]

File: babybuddy/Dockerfile [hunk 2]
```diff
@@ -49,13 +49,16 @@ RUN \
...
-RUN \
-  echo "**** installing ****" && \
-  cd /app/babybuddy && \
-  pip3 install -U --no-cache-dir \
-  pip && \
-  pip install -U --ignore-installed -r requirements.txt
+# Define PATH_MAX before including necessary headers
+RUN echo '#define PATH_MAX 4096' >> /usr/include/limits.h
 
+# Install dependencies and pandas
```
Comment (cdubz) [inline babybuddy/Dockerfile]:
```suggestion
# Install dependencies
```

**Review by cdubz — CHANGES_REQUESTED** [review]

**Comment by OttPeterR** [comment]
@cdubz 
it's still trying to build pandas on its own because of another dependency, here's the relevant part of the logs. Looks like it's becasue of `tablib` I think? Then it'll try and build it and error on its own. [Full logs here](https://github.com/OttPeterR/addon-babybuddy/actions/runs/11696253524/job/32573033828?pr=73). Is this a necessary dependency for BB?
```
#11 33.20 Collecting pandas (from tablib[all,html,ods,xls,xlsx,yaml]==3.7.0->-r requirements.txt (line 39))
#11 33.21   Downloading pandas-2.2.3.tar.gz (4.4 MB)
#11 33.23      ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 4.4/4.4 MB 198.5 MB/s eta 0:00:00
#11 33.88   Installing build dependencies: started
#11 45.40   Installing build dependencies: finished with status 'done'
#11 45.40   Getting requirements to build wheel: started
#11 45.55   Getting requirements to build wheel: finished with status 'done'
#11 45.55   Installing backend dependencies: started
#11 47.35   Installing backend dependencies: finished with status 'done'
#11 47.35   Preparing metadata (pyproject.toml): started
#11 109.9   Preparing metadata (pyproject.toml): still running...
#11 172.8   Preparing metadata (pyproject.toml): still running...
#11 235.2   Preparing metadata (pyproject.toml): still running...
#11 298.4   Preparing metadata (pyproject.toml): still running...
#11 322.2   Preparing metadata (pyproject.toml): finished with status 'error'
#11 322.2   error: subprocess-exited-with-error
#11 322.2   
#11 322.2   × Preparing metadata (pyproject.toml) did not run successfully.
#11 322.2   │ exit code: 1
#11 322.2   ╰─> [210 lines of output]
#11 322.2       + meson setup /tmp/pip-install-_u7ttnot/pandas_89d1369b554d4b43b53f786e3afca13b /tmp/pip-install-_u7ttnot/pandas_89d1369b554d4b43b53f786e3afca13b/.mesonpy-wcmmbazu/build -Dbuildtype=release -Db_ndebug=if-release -Db_vscrt=md --vsenv --native-file=/tmp/pip-install-_u7ttnot/pandas_89d1369b554d4b43b53f786e3afca13b/.mesonpy-wcmmbazu/build/meson-python-native-file.ini
#11 322.2       The Meson build system
#11 322.2       Version: 1.2.1
#11 322.2       Source dir: /tmp/pip-install-_u7ttnot/pandas_89d1369b554d4b43b53f786e3afca13b
#11 322.2       Build dir: /tmp/pip-install-_u7ttnot/pandas_89d1369b554d4b43b53f786e3afca13b/.mesonpy-wcmmbazu/build
#11 322.2       Build type: native build
#11 322.2       Project name: pandas
#11 322.2       Project version: 2.2.3
#11 322.2       C compiler for the host machine: cc (gcc 13.2.1 "cc (Alpine 13.2.1_git20240309) 13.2.1 20240309")
#11 322.2       C linker for the host machine: cc ld.bfd 2.42
#11 322.2       C++ compiler for the host machine: c++ (gcc 13.2.1 "c++ (Alpine 13.2.1_git20240309) 13.2.1 20240309")
#11 322.2       C++ linker for the host machine: c++ ld.bfd 2.42
#11 322.2       Cython compiler for the host machine: cython (cython 3.0.11)
#11 322.2       Host machine cpu family: x86
#11 322.2       Host machine cpu: i686
#11 322.2       Program python found: YES (/usr/bin/python3)
#11 322.2       Found pkg-config: /usr/bin/pkg-config (2.2.0)
#11 322.2       Run-time dependency python found: YES 3.12
#11 322.2       Build targets in project: 53
#11 322.2       
#11 322.2       pandas 2.2.3
#11 322.2       
#11 322.2         User defined options
#11 322.2           Native files: /tmp/pip-install-_u7ttnot/pandas_89d1369b554d4b43b53f786e3afca13b/.mesonpy-wcmmbazu/build/meson-python-native-file.ini
#11 322.2           buildtype   : release
#11 322.2           vsenv       : True
#11 322.2           b_ndebug    : if-release
#11 322.2           b_vscrt     : md
#11 322.2       
#11 322.2       Found ninja-1.11.1.git.kitware.jobserver-1 at /tmp/pip-build-env-b7ric_0v/normal/bin/ninja
#11 322.2       
#11 322.2       Visual Studio environment is needed to run Ninja. It is recommended to use Meson wrapper:
#11 322.2       /tmp/pip-build-env-b7ric_0v/overlay/bin/meson compile -C .
#11 322.2       + /tmp/pip-build-env-b7ric_0v/normal/bin/ninja
#11 322.2       [1/151] Generating pandas/_libs/algos_common_helper_pxi with a custom command
#11 322.2       [2/151] Generating pandas/_libs/hashtable_class_helper_pxi with a custom command
#11 322.2       [3/151] Generating pandas/_libs/khash_primitive_helper_pxi with a custom command
#11 322.2       [4/151] Generating pandas/__init__.py with a custom command
```

**Comment by cdubz** [comment]
Hm. Panda shouldn’t be a dependency anymore. It did used to be. I’ll have to take a closer look… it’s not in the requirements or Pipenv lock file I get locally.

None of those 32bit architectures has a pandas build so this is always going to fail on them if it’s trying to get it for some reason 🤔

**Comment by OttPeterR** [comment]
ChatGPT thinks it's coming from tablib with the xlsx extra dependency. 
edit: just the xlsx extra dep, not tablib on its own

So I think it's this: https://github.com/babybuddy/babybuddy/blob/master/requirements.txt#L39
That sounds like a data import or export feature related dependency.

**Comment by cdubz** [comment]
Yeah I just narrowed that down as well. I don’t know if it comes with xlsx but it definitely comes from all which is included there. It’s odd that I don’t get it locally though… I’ll look closer at it.

**Comment by cdubz** [comment]
Ultimately it comes from django-import-export. This is what I need to review: https://django-import-export.readthedocs.io/en/latest/installation.html#installation-and-configuration

**Comment by cdubz** [comment]
@OttPeterR OK bump Baby Buddy to v2.6.2  and see what ya get!

**Comment by OttPeterR** [comment]
it worked!
I'll merge that other PR into this one then make a new release

**Pull request merged**